### PR TITLE
Add GitHub templates

### PR DESCRIPTION
**Pull request by kinow**
Copied from Cylc Flow. Didn't find anything that actually needed changes, no mentions of Cylc Flow repository directory. :+1:

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+- [ ] Does not contain off-topic changes (use other PRs for other changes).
+<!-- choose one: -->
+- [ ] Appropriate tests are included (unit and/or functional).
+- [ ] Already covered by existing tests.
+- [ ] Does not need tests (why?).
+<!-- choose one: -->
+- [ ] Appropriate change log entry included.
+- [ ] No change log entry required (why? e.g. invisible to users).
+<!-- choose one: -->
+- [ ] (master branch) I have opened a documentation PR at cylc/cylc-doc/pull/XXXX.
```
Comment (hjoliver) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- [ ] I have opened a documentation PR at cylc/cylc-doc/pull/XXXX.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+- [ ] (7.8.x branch) I have updated the documentation in this PR branch.
```
Comment (hjoliver) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
```

**Comment by kinow** [comment]
Doh, missed that 7.8.x branch! Thanks @hjoliver

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+- [ ] Does not contain off-topic changes (use other PRs for other changes).
+<!-- choose one: -->
+- [ ] Appropriate tests are included (unit and/or functional).
+- [ ] Already covered by existing tests.
+- [ ] Does not need tests (why?).
+<!-- choose one: -->
+- [ ] Appropriate change log entry included.
+- [ ] No change log entry required (why? e.g. invisible to users).
+<!-- choose one: -->
+- [ ] I have opened a documentation PR at cylc/cylc-doc/pull/XXXX.
```
Comment (hjoliver) [inline .github/PULL_REQUEST_TEMPLATE.md]: (I guess we don't really have any UI documentation yet - but can worry about that later)

**Pull request merged**